lib/controller.js

```diff
@@ -705,39 +710,152 @@ controller.disableAccessPoint = function(opts) {
+
+    // error handling
+    ch.once('error', function(e) {
+      if (e === 255) {
+        controller.clear()
+          .then(function() {
+            logs.warn('Login to Tessel at failed with KEY ' + opts.key + ' !');
+          }).then(resolve);
+      } else {
+        reject('Error while connected to ' + tessel.name + ':', e);
+      }
+    });
+
+    // exit handling
+    ch.once('exit', function(e) {
+      if (e === 127 || e === 0) { // exit by user
+        // clear console
+        controller.clear()
+          .then(function() {
+            logs.warn('Connection to ' + tessel.name + ' closed!\n');
+            resolve();
+          });
+      } else if (e === 255) {
+        controller.clear()
+          .then(function() {
+            logs.warn('You are not authorized with key ' + opts.key);
+          }).then(resolve);
+      } else {
+        controller.clear()
+          .then(function() {
+            logs.warn('Connection to ' + tessel.name + ' closed due to reason (error code): ', e);
+            reject(e);
+          });
+      }
+    });
+  });
+};
+
 /*
   The T2 root command is used to login into the Tessel's root shell.
 */
 controller.root = function(opts) {
+
   // Only give us LAN connections
   opts.lan = true;
+
   // We must already be authorized
   opts.authorized = true;
-  // Fetch a Tessel
-  return controller.standardTesselCommand(opts, function(tessel) {
-    logs.info('Starting SSH Session on Tessel. Type \'exit\' at the prompt to end.');
+
+  // by --ip tessels outside the netmask are also accessable if key is valid
+  if (opts.ip) {
+    debug('Using IP ' + opts.ip + ' instead of hostname!');
+
+    // remote connection expected (outside netmask)
     return new Promise(function(resolve, reject) {
-      // Spawn a new SSH process
-      var child = cp.spawn('ssh', ['-i',
-        // Use the provided key path
-        opts.key,
-        // Connect to the Tessel's IP Address
-        'root@' + tessel.lanConnection.ip
-      ], {
-        // Pipe our standard streams to the console
-        stdio: 'inherit'
-      });
+      var data = {
+        ip: opts.ip,
+        host: opts.name || opts.ip,
+        port: opts.port,
+        auth: {
+          host: opts.ip,
+          username: opts.username || 'root',
+          passphrase: opts.passphrase || '',
+          privateKey: fs.readFileSync(opts.key),
+          readyTimeout: 5000
+        },
+        connectionType: 'LAN',
+        authorized: false,
+        ssh: undefined
+      };
+      var conn = new lan.LAN.Connection(data);
+      var tessel = new Tessel(conn);
+      tessel.connection.open()
+        .then(function() {
+          if (tessel.lanConnection.authorized) {
+            tessel.getName()
+              .then(function() {
+                controller.clear().then(function() {
+                  controller.remote(tessel, opts)
+                    .then(resolve).catch(function() {
+                      resolve();
+                    });
```

Comment:
@LinusU I would love to use more *ES6* style (for example using `let` instead of `var` all the time). The argument against this would be it is still a minority is familiar with this I assume. But I would prefer this as I said :+1: